modules/core/arrow-core-data/src/main/kotlin/arrow/core/Either.kt

```diff
@@ -367,3 +367,16 @@ fun <A, B> B?.rightIfNotNull(default: () -> A): Either<A, B> = when (this) {
   null -> Either.Left(default())
   else -> Either.Right(this)
 }
+
+
+/**
+ * Applies the given function `f` if this is a [Left], otherwise returns this if this is a [Right].
+ * This is like `flatMap` for the exception.
```

Comment:
instead of saying "exception" we could mention Left:
`This is like `flatMap` for the [Left].`
but the name mentions that this is an error type with "handleErrorWith" so I left the exception there.